Develop

Nuhvi opened the pull request.

Nuhvi committed:
Setup folders, stickler and main.rb

Nuhvi committed:
fix authors links

Nuhvi committed:
adding board.rb

Nuhvi committed:
add private methods row_inedx, col_index and ..

Nuhvi committed:
finished board.rb

tundeiness committed:
created game file and required logic

tundeiness committed:
deleted unneccesary files

Nuhvi committed:
better naming

Nuhvi committed:
Update README.md

tundeiness committed:
Update README.md

Nuhvi committed:
start the engine branch

Nuhvi committed:
start engine done

Nuhvi committed:
Merge pull request #2 from tundeiness/board

Merge Board branch to Develop branch

Nuhvi committed:
merge develop to game

Nuhvi committed:
small changes on description and tests

Nuhvi committed:
Merge branch 'develop' into engine

Nuhvi committed:
Merge pull request #3 from tundeiness/game

Game

Nuhvi committed:
Merge branch 'engine' into develop

Nuhvi committed:
start engine branch

Nuhvi committed:
update main.rb with basic structure

Nuhvi committed:
updated engine.rb with get_cell prompt function

tundeiness committed:
swithced to proper branch

tundeiness committed:
copied from game branch

tundeiness committed:
made alterations to main.rb

tundeiness committed:
enforced a logic to prevent overwriting on input made by a previous player

Nuhvi committed:
rubocop changes

Nuhvi committed:
Fixed a bug with board.rb and added wins & draws

Nuhvi committed:
fixed a bug in prompting a cell number

Nuhvi committed:
finished engine without UI

tundeiness committed:
initiated User interface file

Nuhvi committed:
added display_instructions & display board methods

Nuhvi committed:
added UI.game_over and UI.draw

tundeiness committed:
Merge pull request #4 from tundeiness/engine

Full game logic

Nuhvi committed:
update

Nuhvi committed:
Merge branch 'develop' into ui

Nuhvi committed:
update UI before score

Nuhvi committed:
started player class

Nuhvi committed:
update UI integration in game.rb

tundeiness committed:
added git ignore file to the develop branch to take care of vscode folder that keeps appearing all the time

tundeiness committed:
Merge branch 'develop' of github.com:tundeiness/Tic-Tic-Toe into develop

Nuhvi committed:
deleted engine, updated game  w/ its methods

Nuhvi committed:
added score update

Nuhvi committed:
finished coloring inputs

Nuhvi committed:
upadte README.md

Nuhvi committed:
Update README.md

Nuhvi committed:
Update README.md

Nuhvi committed:
Merge pull request #5 from tundeiness/ui

Ui

Nuhvi committed:
Merge pull request #6 from tundeiness/doc

Doc

Nuhvi committed:
Update README.md

Nuhvi committed:
Fix the prompt message frmo X/Y to X/O

maelfosso commented on lib/game.rb:
```diff
@@ -0,0 +1,84 @@
+      answer = prompt(game_over + winlose + prompt_play_again)
+    end
+    new_match if answer == 'y'
+    play if answer == 'y'
```
`new_match` method is already calling the `play` method at the end of his execution.
Unless I'm wrong, why did you put that line?

Nuhvi replied:
Yes, you are right, this should be omitted, working on it, thanks for your great feedback!

maelfosso replied:
OK. Once done, fill the project completion form.

maelfosso commented:
Hello coders,

Please, could you check the comment above?

Nuhvi committed:
small fixes

(1..9).to_a. instead if [1,2,3,...,9]
@turn.name instead of coounted the marked cells
omitted unnecessary step in Game #play-again? method

used global variables in diagonal helpers

commented the colors in the Player class for future debugging.

added update _turn method to further simplify the play method

Nuhvi committed:
Update README.md
